README.md

```diff
@@ -1 +1,47 @@
-## Canvas Course Manager
\ No newline at end of file
+## Canvas Course Manager
+
+## Development
```

Comment:
I would really like to hear, the development set up is running on 2 container (client and server) and prod doesn't seems to that way

This setup seems to be new to me and I cannot connect why client and server running on 2 different instances. what is the purpose they needs to be running in separate ports?

Reply:
We can discuss. Maybe there is another way to do this. The separate servers here I believe are most to allow us to leverage `react-scripts` and its reloading-for-dev and building-for-prod functionality. I was mostly seeing this pattern when I researched online, but I could have been missing something.

Yes, the production workflow is different, the Express application serves an optimized build as static assets.

Reply:
I see this [setup](https://levelup.gitconnected.com/how-to-setup-environment-using-react-webpack-express-babel-d5f1b572b678) with express and React using webpack ( with out docker though ) https://github.com/Jatin-8898/express-react-webpack-babel-setup

Reply:
Thanks, I'll take a look. I think if I go through the `webpack` tutorial, I can probably do something like this without too much trouble. `react-scripts` (`create-react-app`) does seem to handle a lot of this stuff for you, but maybe with a lean setup like this, it's not needed, and we can limit development to one server.

Reply:
I think Babel is not needed with Type script since ts-loader handles both types and transpiling. I believe, react-script is convenient but I feel it has lot of stuff and it is concealing what is going on underneath.

I think we still 2 watchers in Dev mode one looking at the express(BE) changes,`nodemon` is probably doing that. One looking at React changes (FE). The interplay will be interesting to see like how the react bundle changes noticed by BE. If BE reloads after a FE bundle change it is bit unclear right now.

Reply:
See my latest commits. FE changes will be registered by `webpack`. BE changes will trigger a full reset, including a new `webpack` bundling process, but I'm not sure if we can avoid that given the current one server configuration.